src/1-Profile/Edit-Profile.tsx

```diff
@@ -30,66 +31,64 @@ const EditProfile = ({navigation}:StackNavigationProps):JSX.Element => {
     const RequestAccountHeader = {
       headers: {
         "jwt": jwt, 
-        "userID": userID,
       }
     }
-    // store user profile data as an indexable object
-    const userProfileValues: Record<string, string> = {};
-    for (const [key, value] of Object.entries(userProfile)) {
-      userProfileValues[key] = value;
-    }
 
-    // assign user profile data as default values for fields
-    const inputFieldJSON: Record<string, string> = {};
-    EDIT_PROFILE_FIELDS.forEach((field) => {
-      if (userProfileValues[field.field]) inputFieldJSON[field.field] = userProfileValues[field.field];
-    })
+    const createFormValues = ():Record<string, string> => {
+      const formValues: Record<string, string> = {};
+      EDIT_PROFILE_FIELDS.forEach((field) => {
+        //@ts-ignore
+        formValues[field.field] = userProfile[field.field] || field.value;
+      });
+      return formValues;
+    }
 
     const {
       control,
       handleSubmit,
       formState: { errors },
       clearErrors,
     } = useForm({
-      defaultValues: inputFieldJSON
+      defaultValues: createFormValues()
     });
 
-    const onEditFields = (fieldData:any):void => {
+    const onEditFields = (formValues:Record<string, string>):void => {
 
       // filter out fields that didn't change
-      const editedFields: Record<string, string> = {};
-      for (const [key, value] of Object.entries(fieldData)) {
-        if (value !== userProfileValues[key]) editedFields[key] = value as unknown as string;
+      const editedFields = {} as ProfileEditRequestBody;
+      for (const [key, value] of Object.entries(formValues)) {
+        //@ts-ignore
+        if (value !== userProfile[key]) editedFields[key] = value;
       }
-
       // send data to server
       axios.patch(`${DOMAIN}/api/user/` + userProfile.userID, editedFields, RequestAccountHeader,
         ).then(response => {
             console.log("Profile edit success.");
-
-            // IMPORTANT: The following assumes that this route will return the user's userRole, userRoleList, and walkLevel by default, and other fields that changed in addition.
-            // IF THIS CHANGES, EDIT PROFILE WILL BREAK!
-            if (Object.values(response.data).length > 3) {
-              // something other than the password changed, and we need to save it to redux
-              for (const [key, value] of Object.entries(userProfileValues)) {
-                if (response.data[key]) userProfileValues[key] = response.data[key];
-              }
+            var updatedUserProfile = {...userProfile}
+            var profileChange = false;
+            for (const [key, value] of Object.entries(editedFields)) {
+              //@ts-ignore - Only copy over InputFields that exist in ProfileResponse
+              if (updatedUserProfile[key]) {
```

Comment:
Maybe better to compare against `undefined`, since some fields are boolean that may be in the `false` state.